src/services/export/XLSExporterForObservations.ts

```diff
@@ -1,20 +1,86 @@
-import path from 'path';
+import stream from 'stream';
 import { NextFunction, Request, Response } from 'express';
-// import { getRepository, Repository } from 'typeorm';
+import { getRepository, Repository } from 'typeorm';
+import { write, utils } from 'xlsx';
 import AbstractExporter, { ExporterType } from './AbstractExporter';
-// import { Observation } from '../../entities/observation-entity';
+import { Observation } from '../../entities/observation-entity';
 
 export default class XLSExporterForObservations extends AbstractExporter {
   public type: ExporterType = 'XLS';
 
   public route: string = 'observations';
 
-  // private observations: Repository<Observation> = getRepository(Observation);
+  private observations: Repository<Observation> = getRepository(Observation);
+
+  private ObservationColumns = [
+    'ringMentioned',
+    'photos',
+    'distance',
+    'direction',
+    'elapsedTime',
+    'colorRing',
+    'date',
+    'geographicalCoordinates',
+    'placeName',
+    'remarks',
+    'verified',
+  ];
+
+  private ObservationColumnsByDesc = [
+    'sexMentioned',
+    'sexConcluded',
+    'ageMentioned',
+    'ageConcluded',
+    'manipulated',
+    'movedBeforeTheCapture',
+    'catchingMethod',
+    'catchingLures',
+    'accuracyOfDate',
+    'accuracyOfCoordinates',
+    'status',
+    'pullusAge',
+    'accuracyOfPullusAge',
+    'condition',
+    'circumstances',
+    'circumstancesPresumed',
+  ];
 
-  /* eslint-disable-next-line class-methods-use-this */
   public async export(_req: Request, res: Response, next: NextFunction): Promise<void> {
     try {
-      res.sendFile(path.resolve(__dirname, './sample.xls'));
+      const lang = 'desc_eng';
+      const qb = this.observations.createQueryBuilder('observation');
+      const withSelect = qb.select([
+        'ring.id',
+        'finder.firstName AS "finder_firstName"',
+        'finder.lastName AS "finder_lastName"',
+        'finder.role AS "finder_role"',
+        'speciesMentioned.species',
+        `speciesMentioned.${lang} AS speciesMentioned_desc`,
+        'speciesConcluded.species',
+        `speciesConcluded.${lang} AS speciesMentioned_desc`,
+        ...this.ObservationColumns.map(column => `observation.${column} AS "${column}"`),
+        ...this.ObservationColumnsByDesc.map(column => `${column}.${lang} AS "${column}"`),
+      ]);
+
+      const withJoins = this.ObservationColumnsByDesc.reduce(
+        (acc, column) => acc.leftJoin(`observation.${column}`, column),
+        withSelect,
+      )
+        .leftJoin('observation.ring', 'ring')
+        .leftJoin('observation.finder', 'finder')
+        .leftJoin('observation.speciesMentioned', 'speciesMentioned')
+        .leftJoin('observation.speciesConcluded', 'speciesConcluded');
```

Comment:
too odd request I would able to do same just with simple find and `eager` or manually specify needed relations

Then on stage of constructing xls, you can remove unneeded data like `desc_rus` or what else.

Reply:
I rewrote 'query builder' with 'find'